### DISCUSSION TITLE
Is there a way to use locally deployed AIs?

### DISCUSSION THREAD

**Discussion by yuanhan** [apirrone/Memento#75] [Ideas]
like a simple way to connect with gpt4all?
maybe share the data dir to gpt4all?

**Comment by apirrone** [comment]
Hi! 

Memento uses `LangChain` to wrap LLM queries, so it should be quite straightforward to use different LLMs, including local ones !

You would need to change `OpenAIEmbeddings` and `ChatOpenAI`, below are the lines you would need to edit in the code : 

- https://github.com/apirrone/Memento/blob/513a80c2ba155115dfcea0bc808b0c41a1fd4380/memento/timeline/chat.py#L60
- https://github.com/apirrone/Memento/blob/513a80c2ba155115dfcea0bc808b0c41a1fd4380/memento/timeline/chat.py#L86
- https://github.com/apirrone/Memento/blob/513a80c2ba155115dfcea0bc808b0c41a1fd4380/memento/background.py#L58
